### ISSUE TITLE
docker build pullSecret broken in 1.1.2

### ISSUE THREAD

**Issue by vbehar** [openshift/origin#7228]
Hi,

we found a serious regression in 1.1.2 when building using the docker strategy and a pullSecret (to pull a private docker image as the base image), related to the new support of the .docker/config.json format : 

previously, only the .dockercfg format was supported, and so the `PULL_DOCKERCFG_PATH` env var in the builder pod was set to `/var/run/secrets/openshift.io/pull/.dockercfg`

but now, the `PULL_DOCKERCFG_PATH` env var in the builder pod is set to `/var/run/secrets/openshift.io/pull` but the code reading it has not changed, so it fails with the following error:

```
builder.go:202] Error: build error: unexpected end of JSON input
```

It's a big blocker for us, because we have quite a few projects using private base images for building...

**Comment by bparees** [comment]
Pull the new sti-builder and docker builder image.  It sounds like you have mismatched levels. You should have a warning in your build log about that.

**Comment by vbehar** [comment]
Ok I'll check that later (can't right now). but in the source code we found https://github.com/openshift/origin/blob/v1.1.2/pkg/build/builder/docker.go#L237

**Comment by bparees** [comment]
@vbehar i think you're correct, it looks like the docker codepath wasn't updated when the s2i codepath was updated.  will work on a fix.

@smarterclayton fyi.

**Comment by smarterclayton** [comment]
Do we have enough tests around pull secrets?

**Comment by bparees** [comment]
@smarterclayton clearly not, i'm working on adding one as part of the fix for this.

**Comment by danni** [comment]
Is there a way to upgrade the builder? To something with the fix? I couldn't seem to find one.

**Comment by danni** [comment]
Or is there a way to specify what builder image to use that isn't pulling the image down on each node and forcing the replacement tag? Missing from the docs?

**Comment by vbehar** [comment]
@bparees do you plan to release a 1.1.2.1 or something like that with this fix ? It would be awesome ;-) Otherwise we are stuck on 1.1.1 until the next release (and there are features we have been waiting for in 1.1.2 !!)

**Comment by bparees** [comment]
@danni you can docker pull the latest openshift/origin-docker-builder image to your cluster nodes and run openshift start with "--latest-images" to allow it to use the latest image that exists on your node.

@vbehar that's a question for @smarterclayton.  But it definitely seems like we should given the severity of this issue.  Or you can do what i described above to grab the new image.

(you could also grab the new image and then re-tag it w/ the name o the 1.1.2 image, instead of specifying --latest images, but that is definitely sketchy)

**Comment by smarterclayton** [comment]
This has been fixed in 1.1.3

**Comment by vbehar** [comment]
yes I saw that, thanks!
